tests/TestCase/Validation/ValidationRuleTest.php

```diff
@@ -178,4 +180,14 @@ public function testGet(): void
         $this->assertSame('bar', $Rule->get('message'));
         $this->assertEquals(['param'], $Rule->get('pass'));
     }
+
+    public function testAddDuplicateName(): void
+    {
+        $rules = new ValidationSet();
+        $rules->add('myUniqueName', ['rule' => fn () => false]);
+
+        $this->expectException(CakeException::class);
+        $rules->add('myUniqueName', ['rule' => fn () => true]);
+        $this->fail('Exception not thrown');
```

Comment:
Is this fail() call necessary? I am pretty sure the test will fail if the exceptException() isn't met.

Reply:
It's not. Removing it